### PR TITLE
[TACHYON-985] Modify TachyonFileSystem to use non-recursive methods by default

### PR DESCRIPTION
**Pull request by chen-tian**
https://tachyon.atlassian.net/browse/TACHYON-985

### CODE REVIEW

**Comment by haoyuan** [comment]
Jenkins, test this please.

File: shell/src/main/java/tachyon/shell/TFsShell.java [hunk 1]
```diff
@@ -932,7 +932,7 @@ public int touch(TachyonURI path) throws IOException {
    */
   public int unpin(TachyonURI path) throws IOException {
     try {
-      mTfs.setPin(mTfs.open(path), false);
+      mTfs.setPin(mTfs.open(path), true);
```
Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/TFsShell.java]: Why is this changed to true? Also I think you can use the `unpin` method which is shorthand for this operation.

Comment (chen-tian) [inline shell/src/main/java/tachyon/shell/TFsShell.java]: @calvinjia Because the java doc comment of this function says this function work recursively on a folder. We need to make a decision on this function's behavior.

Comment (calvinjia) [inline shell/src/main/java/tachyon/shell/TFsShell.java]: @chen-tian Oh I think there is some confusion here, the setPin command's second argument is to pin or unpin (true is to pin, false is to unpin). I think pin's behavior is always recursive.

Comment (chen-tian) [inline shell/src/main/java/tachyon/shell/TFsShell.java]: @calvinjia  yeah you are right.

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java [hunk 2]
```diff
@@ -213,7 +214,7 @@ public FileOutStream getOutStream(long fileId, ClientOptions options) throws IOE
    */
   public boolean mkdirs(TachyonURI path) throws FileAlreadyExistException, InvalidPathException,
       IOException {
-    return mkdirs(path, true);
+    return mkdirs(path, false);
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: rename this to mkdir

File: integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java [hunk 3]
```diff
@@ -153,8 +153,8 @@ public void getFileStatusTest() throws IOException, TException {
   public void mkdirTest() throws IOException, TException {
     String uniqPath = PathUtils.uniqPath();
     for (int k = 0; k < 10; k ++) {
-      Assert.assertTrue(sTfs.mkdirs(new TachyonURI(uniqPath + k)));
-      Assert.assertTrue(sTfs.mkdirs(new TachyonURI(uniqPath + k)));
+      Assert.assertTrue(sTfs.mkdirs(new TachyonURI(uniqPath + k), true));
```
Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java]: Define the following constant in TachyonFileSystem:

`` public static final RECURSIVE = true;``

and use it everywhere to improve the code readability:

`` Assert.assertTrue(sTfs.mkdirs(new TachyonURI(uniqPath + k), TachyonFileSystem.RECURSIVE));``

Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java]: Similarly for other instances of the recursive invocations.

**Comment by jsimsa** [comment]
Left some comments for you.

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java (hunk 2 above)
Comment (chen-tian) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: @jsimsa @calvinjia  Created a new issue 1038 for this as there are many "mkdirs" in the code base. We need to decide whether we refactor all of them or just the convenient function in this file.

File: integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java (hunk 3 above)
Comment (chen-tian) [inline integration-tests/src/test/java/tachyon/client/TachyonFileSystemIntegrationTest.java]: done.

**Comment by chen-tian** [comment]
@jsimsa I think it's better to handle the function renaming in different PR. I've created a ticket for it. The second comment is addressed. I like it.

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java (hunk 2 above)
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: Sounds good. Add the following TODO for the mkdirs function:

``TODO(calvin,jiri): Consider renaming to mkdir``

**Comment by jsimsa** [comment]
Thanks. One last minor comment from me.

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java (hunk 2 above)
Comment (chen-tian) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: @jsimsa  Addressed.

**Comment by jsimsa** [comment]
LGTM +1

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java [hunk 4]
```diff
@@ -99,19 +101,20 @@ public long createEmptyFile(TachyonURI path, ClientOptions options) throws IOExc
...
+   * Convenience method for delete without recursive set. This is the same as calling delete(file,
+   * false).
    *
    * @param file the handler for the file to delete recursively
    * @throws FileDoesNotExistException if the file does not exist in Tachyon space
    * @throws IOException if the master cannot delete the file
    */
   public void delete(TachyonFile file) throws FileDoesNotExistException, IOException {
-    delete(file, true);
+    delete(file, false);
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: Should these `false` be `!RECURSIVE`?

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
LGTM
